specifyweb/frontend/js_src/lib/components/Statistics/utils.tsx

```diff
@@ -0,0 +1,154 @@
+          fields: fields.map(({ path, ...field }, index) =>
+            serializeResource(
+              makeQueryField(tableName, path, { ...field, position: index })
+            )
+          ),
+        })
+      ),
+    [tableName, fields]
+  );
+}
+export function useFrontEndStat(
+  query: SpecifyResource<SpQuery>
+): string | undefined {
+  const [countReturn] = useAsyncState(
+    React.useCallback(
+      async () =>
+        ajax<{
+          readonly count: number;
+        }>('/stored_query/ephemeral/', {
+          method: 'POST',
+          headers: {
+            // eslint-disable-next-line @typescript-eslint/naming-convention
+            Accept: 'application/json',
+          },
+          body: keysToLowerCase({
+            ...serializeResource(query),
+            countOnly: true,
+          }),
+        }).then(({ data }) => formatNumber(data.count)),
+      [query]
+    ),
+    false
+  );
+  return countReturn;
+}
+
+export type BackendStatsResult = {
+  readonly holdings: {
+    readonly familiesRepresented: number;
+    readonly generaRepresented: number;
+    readonly speciesRepresented: number;
+  };
+  readonly preparations: IR<{
+    readonly lots: number;
+    readonly total: number;
+  }>;
+  readonly localityGeography: { readonly countries: number };
+  readonly typeSpecimens: IR<number>;
+};
+
+export type QueryBuildStat = State<
+  'Querybuildstat',
+  {
+    readonly tableName: keyof Tables;
+    readonly fields: RA<
+      Partial<SerializedResource<SpQueryField>> & { readonly path: string }
+    >;
+  }
+>;
+
+export type BackendStat = State<
+  'Backendstat',
+  { readonly value: number | string | undefined }
+>;
+
+export type StatItemSpec = BackendStat | QueryBuildStat;
+
+export function StatsResult({
+  statValue,
+  query,
+  statLabel,
+  onClick: handleClick,
+  onRemove: handleRemove,
+}: {
+  readonly statValue: string | number | undefined;
+  readonly query: SpecifyResource<SpQuery> | undefined;
+  readonly statLabel: string;
+  readonly onClick: (() => void) | undefined;
+  readonly onRemove: (() => void) | undefined;
+}): JSX.Element {
+  const [isOpen, handleOpen, handleClose] = useBooleanState();
+  return (
+    <>
+      {statLabel === undefined ? (
+        <div>{commonText('loading')}</div>
+      ) : (
+        <p className="flex gap-2">
+          {typeof handleRemove === 'function' && (
+            <Button.Icon
+              icon="trash"
+              title={commonText('remove')}
+              onClick={handleRemove}
+            />
+          )}
+          <Button.LikeLink
+            className="flex-1"
+            onClick={
+              handleClick ?? (query === undefined ? undefined : handleOpen)
+            }
+          >
+            <span>{statLabel}</span>
+            <span className="-ml-2 flex-1" />
+            <span>{statValue ?? commonText('loading')}</span>
+          </Button.LikeLink>
+        </p>
+      )}
+
+      {isOpen && query !== undefined && (
```

Comment:
Use a ternary here rather than the `&&` trick - that trick should only be used when there is a single condition. Otherwise, it becomes too hard to comprehend